File: backend/src/main/java/com/legymernok/backend/model/sector/Sector.java
package com.legymernok.backend.model.sector;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "sectors")
public class Sector {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String iconUrl;

    @Column(nullable = false)
    @Builder.Default
    private Integer orderIndex = 0;

    @CreationTimestamp
    @Column(updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;
}

File: frontend/src/types/sector.ts
export interface SectorResponse {
  id: string;
  name: string;
  description: string | null;
  iconUrl: string | null;
  orderIndex: number;
  starSystemCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSectorRequest {
  name: string;
  description?: string;
  iconUrl?: string;
}
